File: 953-Verifying an Alien Dictionary.py
class Solution(object):
    def isAlienSorted(self, words, order):
        """
        :type words: List[str]
        :type order: str
        :rtype: bool
        """
        hmap = {c:i for i,c in enumerate(order)}
        
        for word1, word2 in zip(words, words[1:]):
            i=0
            while i < len(word1) and i < len(word2):
                if hmap[word1[i]] < hmap[word2[i]]:
                    break
                elif hmap[word1[i]] > hmap[word2[i]]:
                    return False
                else:
                    i+=1
            if i == len(word2) and word1[i:]:
                return False
            
        return True

File: JavaScript/953-Verifying an Alien Dictionary.js
/**
 * @param {string[]} words
 * @param {string} order
 * @return {boolean}
 */
var isAlienSorted = function(words, order) {
    
    let hmap = new Map();
    
    for (let j=0; j<order.length; j++){
        hmap.set(order[j], j);
    }
    
    for (let i=1; i < words.length; i++){
        let word1 = words[i-1];
        let word2 = words[i];
        let ind = 0;
        while (ind < word1.length && ind < word2.length){
            if (hmap.get(word1[ind]) < hmap.get(word2[ind])) break;
            else if (hmap.get(word1[ind]) > hmap.get(word2[ind])) return false;
            else ind++;
        }
        if (ind === word2.length && word1.substr(ind).length > 0) return false;
    }
    return true;
};